### PR TITLE
Swc 2368

### PR DESCRIPTION
**Pull request by Njanderson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/presenter/ChangeUsernamePresenter.java [hunk 1]
```diff
@@ -69,14 +76,15 @@ public void onSuccess(Void result) {
 					
 					@Override
 					public void onFailure(Throwable caught) {
-						if (!DisplayUtils.handleServiceException(caught, globalAppState, authController.isLoggedIn(), view))
-							view.showSetUsernameError(caught);
+						synAlert.handleException(caught);
+						synAlert.showError("Unable to set username. Try again later.");
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ChangeUsernamePresenter.java]: remove showError, as this will lose the original error message (displayed in handleException).

File: src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java [hunk 2]
```diff
@@ -84,11 +88,13 @@ public void start(AcceptsOneWidget panel, EventBus eventBus) {
 	public void setPlace(Synapse place) {
 		this.place = place;
 		this.view.setPresenter(this);		
+		this.view.setSynAlertWidget(synAlert.asWidget());
 		
 		this.entityId = place.getEntityId();
 		this.versionNumber = place.getVersionNumber();
 		this.area = place.getArea();
 		this.areaToken = place.getAreaToken();
+		this.synAlert.clear();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java]: can remove this synAlert.clear()

File: src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java [hunk 3]
```diff
@@ -163,8 +169,9 @@ public void onFailure(Throwable caught) {
 					view.show404();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java:169]: Do a synAlert.clear();  as one of the first lines in refresh() (before async call).

File: src/main/java/org/sagebionetworks/web/client/presenter/PeopleSearchPresenter.java [hunk 4]
```diff
@@ -57,6 +60,8 @@ public void setPlace(PeopleSearch place) {
 		this.place = place;
 		this.view.setPresenter(this);
 		this.view.clear();
+		this.synAlert.clear();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/PeopleSearchPresenter.java]: can remove this synAlert.clear().

File: src/main/java/org/sagebionetworks/web/client/presenter/PeopleSearchPresenter.java [hunk 5]
```diff
@@ -73,7 +78,8 @@ public void search(final String searchTerm, Integer offset) {
 		else
 			this.offset = offset;
 		//execute search, and update view with the results
-		synapseClient.getUserGroupHeadersByPrefix(searchTerm, (long) SEARCH_PEOPLE_LIMIT, (long) this.offset, new AsyncCallback<UserGroupHeaderResponsePage>() {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/PeopleSearchPresenter.java:76]: synAlert.clear() as a first item of search().

File: src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java [hunk 6]
```diff
@@ -213,7 +231,7 @@ public void onSuccess(UserProfile profile) {
 			@Override
 			public void onFailure(Throwable caught) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:233]: profileSynAlert.clear(); before async call to get user profile.

File: src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java [hunk 7]
```diff
@@ -601,9 +619,7 @@ public void onFailure(Throwable caught) {
 				if(caught instanceof ConflictException) {
 					view.showErrorMessage(DisplayConstants.WARNING_PROJECT_NAME_EXISTS);
 				} else {
-					if(!DisplayUtils.handleServiceException(caught, globalApplicationState, authenticationController.isLoggedIn(), view)) {					
-						view.showErrorMessage(DisplayConstants.ERROR_GENERIC_RELOAD);
-					} 
+					projectSynAlert.handleException(caught);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:624]: projectSynAlert.clear() before async call to create project.

File: src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java [hunk 8]
```diff
@@ -630,12 +646,11 @@ public void onFailure(Throwable caught) {
 				if(caught instanceof ConflictException) {
 					view.showErrorMessage(DisplayConstants.WARNING_TEAM_NAME_EXISTS);
 				} else {
-					if(!DisplayUtils.handleServiceException(caught, globalApplicationState, authenticationController.isLoggedIn(), view)) {					
-						view.showErrorMessage(caught.getMessage());
-					}
+					teamSynAlert.handleException(caught);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:652]: teamSynAlert.clear before async call to create team.

File: src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java [hunk 9]
```diff
@@ -662,6 +680,9 @@ private void loggedInCheck() {
 	
 	private void showView(Profile place) {
 		view.clear();
+		profileSynAlert.clear();
+		projectSynAlert.clear();
+		teamSynAlert.clear();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:685]: (resetSynAlertWidgets)

File: src/main/resources/org/sagebionetworks/web/client/view/TrashViewImpl.ui.xml [hunk 10]
```diff
@@ -13,7 +14,8 @@
 			<div class="well col-md-offset-1 col-md-10 col-sm-offset-2 col-sm-8">
 				<h2 style="display: inline;" >Trash Can</h2>
 					<g:Button ui:field="deleteAllButton" styleName="btn btn-danger margin-bottom-5 btn-lg right">Empty Trash</g:Button>
-					<div class="panel panel-body margin-top-30" style="margin-bottom: 5;">
+					<g:SimplePanel ui:field="synAlertPanel" addStyleNames="clearright clearfix"/>
```
Comment (jay-hodgson) [inline src/main/resources/org/sagebionetworks/web/client/view/TrashViewImpl.ui.xml:17]: clearright?

File: src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java (hunk 9 above)
Comment (Njanderson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:685]: Should I put this into another pull request?

Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java:685]: Yes please, but none of the issues I raised are urgent.
